Mathlib/Analysis/BoxIntegral/Basic.lean

```diff
@@ -673,44 +674,169 @@ open MeasureTheory
 ### Integrability conditions
 -/
 
+open Prepartition EMetric
 
 variable (l)
 
+/-- A function that is bounded and a.e. continuous on a box `I` is integrable on `I`. -/
+theorem integrable_of_bounded_and_ae_continuousWithinAt (l : IntegrationParams) [CompleteSpace E]
+    {I : Box ι} {f : ℝⁿ → E} (hb : ∃ C : ℝ, ∀ x ∈ Box.Icc I, ‖f x‖ ≤ C) (μ : Measure ℝⁿ)
+    [IsLocallyFiniteMeasure μ]
+    (hc : ∀ᵐ x ∂(μ.restrict (Box.Icc I)), ContinuousWithinAt f (Box.Icc I) x) :
+    Integrable I l f μ.toBoxAdditive.toSMul := by
```

Comment:
This proof is really long and is split into several `have`. Are you sure you cannot split off lemmas?

Also there are lots of fully qualified names. Can’t you open some name spaces?

Reply:
I opened some more name spaces. I think that's a good change.

I had already thought about trying to split off some lemmas to make the proof shorter, but I don't really see a good place to do it. I think the big `have`s all rely on a lot of specific details of the proof; I think any gain in making the main proof a little shorter would be outweighed by the lemma being somewhat unmotivated out of context. But I'm open to suggestions if you have a good idea.

 I turned the last `have` into a `suffices`, which I think makes the end of the proof a little more readable.